src/charm.py

```diff
@@ -206,6 +219,85 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
         ):
             self.unit.status = ActiveStatus()
 
+    # flake8: noqa: C901
```

Comment:
Suppressing this linter warning feels like a code smell, and this method is a little hefty, but I can't really figure out better ways to implement each part of `_on_db_router_relation_changed()` - is there any part of this that can be extracted into a separate method to encapsulate some of the complicated parts?

Reply:
i tried to refactor it to simplify it in 5a2d6a26ef82b30eb9937885ae5460ab5f22024f